tests/testthat/test-manage_deps.R

```diff
@@ -242,9 +246,23 @@ test_that("manage_deps() restores Python dependencies", {
   skip_on_os("windows")
 
   req_file <- fs::path(lsn, "requirements.txt")
+  if (file.exists(req_file)) fs::file_delete(req_file)
   writeLines("numpy", req_file)
   res <- manage_deps(lsn, quiet = TRUE)
 
   expect_no_error({numpy <- local_load_py_pkg(lsn, "numpy")})
   expect_s3_class(numpy, "python.builtin.module")
 })
+
+
+test_that("update_cache does not remove uninstalled Python dependencies from requirements.txt", {
+  skip_if_not(reticulate_installable, "reticulate is not installable")
+  skip_on_cran()
+  skip_on_os("windows")
+
+  req_file <- fs::path(lsn, "requirements.txt")
+  write("pandas", req_file, append = TRUE, sep = "\n")
+
+  res <- update_cache(lsn, prompt = FALSE, quiet = FALSE)
+  expect_true(any(grepl("pandas", readLines(req_file))))
```

Comment:
Maybe a smaller library (and something we for sure won't use), with a smaller footprint could make this test faster. e.g., [`art`](https://github.com/sepandhaghighi/art).